src/externalized/strategic/CreatureLairModel.cc

```diff
@@ -0,0 +1,259 @@
+#include "CreatureLairModel.h"
+
+#include "Campaign_Types.h"
+#include "Creature_Spreading.h"
+#include "Random.h"
+#include <set>
+
+
+CreatureLairModel::CreatureLairModel(const uint8_t lairId_, const uint8_t associatedMineId_, const uint8_t entranceSector_, const uint8_t entranceSectorLevel_, const std::vector<CreatureLairSector> lairSectors_, const std::vector<CreatureAttackSector> attackSectors_, const uint8_t warpExitSector_, const uint16_t warpExitGridNo_)
+	: lairId(lairId_), associatedMineId(associatedMineId_), 
+	entranceSector(entranceSector_), entranceSectorLevel(entranceSectorLevel_), 
+	lairSectors(lairSectors_), attackSectors(attackSectors_),
+	warpExitSector(warpExitSector_), warpExitGridNo(warpExitGridNo_) {}
+
+uint8_t creatureHabitatFromString(std::string habitat)
+{
+	if (habitat == "QUEEN_LAIR") return QUEEN_LAIR;
+	if (habitat == "LAIR") return LAIR;
+	if (habitat == "LAIR_ENTRANCE") return LAIR_ENTRANCE;
+	if (habitat == "INNER_MINE") return INNER_MINE;
+	if (habitat == "OUTER_MINE") return OUTER_MINE;
+	if (habitat == "FEEDING_GROUNDS") return FEEDING_GROUNDS;
+	if (habitat == "MINE_EXIT") return MINE_EXIT;
+
+	SLOGE(ST::format("Unrecognized creature habitat: '{}'", habitat));
+	throw std::runtime_error("");
+}
+
+InsertionCode insertionCodeFromString(std::string code)
+{
+	if (code == "NORTH") return INSERTION_CODE_NORTH;
+	if (code == "SOUTH") return INSERTION_CODE_SOUTH;
+	if (code == "EAST") return INSERTION_CODE_EAST;
+	if (code == "WEST") return INSERTION_CODE_WEST;
+	if (code == "GRIDNO") return INSERTION_CODE_GRIDNO;
+	if (code == "ARRIVING_GAME") return INSERTION_CODE_ARRIVING_GAME;
+	if (code == "CHOPPER") return INSERTION_CODE_CHOPPER;
+	if (code == "PRIMARY_EDGEINDEX") return INSERTION_CODE_PRIMARY_EDGEINDEX;
+	if (code == "SECONDARY_EDGEINDEX") return INSERTION_CODE_SECONDARY_EDGEINDEX;
+	if (code == "CENTER") return INSERTION_CODE_CENTER;
```

Comment:
Might look nicer if it was all stored in a static std::map, though I can't decide if I care enough. :thinking:

Reply:
I am thinking about a small refactor of all these deserialization logic.

Maybe a `DeserializationUtil` module, which contains all these `enum` mappings, and the `SECTOR_FROM_SECTOR_SHORT_STRING` handling.

Reply:
Sounds fine. C++11 allows us to do it more cleanly